src/appleseed.shaders/src/appleseed/as_triplanar.osl

```diff
@@ -0,0 +1,874 @@
+        string as_maya_attribute_short_name = "zsf",
+        string widget = "checkBox",
+        int as_maya_attribute_connectable = 0,
+        int as_maya_attribute_keyable = 0,
+        int as_maya_attribute_hidden = 1,
+        int gafferNoduleLayoutVisible = 0,
+        string label = "S Flip",
+        string page = "Projection.Z Axis"
+    ]],
+    int in_z_axis_tflip = 0
+    [[
+        string as_maya_attribute_name = "zAxisTFlip",
+        string as_maya_attribute_short_name = "ztf",
+        string widget = "checkBox",
+        int as_maya_attribute_connectable = 0,
+        int as_maya_attribute_keyable = 0,
+        int as_maya_attribute_hidden = 1,
+        int gafferNoduleLayoutVisible = 0,
+        string label = "T Flip",
+        string page = "Projection.Z Axis"
+    ]],
+    string in_z_tex_eotf = "sRGB"
+    [[
+        string as_maya_attribute_name = "zTextureEOTF",
+        string as_maya_attribute_short_name = "zeo",
+        string label = "Input Transfer Function",
+        string page = "Projection.Z Axis.Color Management",
+        string widget = "mapper",
+        string options = "None/Raw|sRGB|Rec.709|Gamma 2.2|Gamma 2.4|Gamma 2.6 (DCI)|Rec.1886|Rec.2020",
+        int as_maya_attribute_connectable = 0,
+        int as_maya_attribute_keyable = 0,
+        int as_maya_attribute_hidden = 1,
+        int gafferNoduleLayoutVisible = 0,
+        string help = "Transfer function of the ingested material, refers to the Opto-Electronic Transfer Function required to transform the input into scene-linear values."
+    ]],
+    string in_z_tex_rgb_primaries = "sRGB/Rec.709"
+    [[
+        string as_maya_attribute_name = "zTextureRGBPrimaries",
+        string as_maya_attribute_short_name = "zpr",
+        string label = "RGB Primaries",
+        string page = "Projection.Z Axis.Color Management",
+        string widget = "popup",
+        string options = "Raw|sRGB/Rec.709|AdobeRGB|Rec.2020|DCI-P3|ACES|ACEScg",
+        int as_maya_attribute_connectable = 0,
+        int as_maya_attribute_keyable = 0,
+        int as_maya_attribute_hidden = 1,
+        int gafferNoduleLayoutVisible = 0,
+        string help = "Texture RGB primaries, note that Rec.709 and sRGB share the same CIE xy chromaticity coordinates."
+    ]],
+    float in_randomization = 0.0
+    [[
+        string as_maya_attribute_name = "randomization",
+        string as_maya_attribute_short_name = "ran",
+        float min = 0.0,
+        float max = 1.0,
+        string label = "Randomization",
+        string page = "Projection.Randomization"
+    ]],
+    int in_manifold = 0
+    [[
+        string as_maya_attribute_name = "manifold",
+        string as_maya_attribute_short_name = "man",
+        string label = "Manifold",
+        string page = "Projection.Randomization",
+        string help = "If a idManifold is connected, then it's used, otherwise the default assembly instance name ID is used instead."
```

Comment:
"name ID"? I can imagine *name* or *ID* but what "name ID" means?

Reply:
Let me rephrase that. It's a hash built from object:assembly_instance_name.